samples/basic-chat/server-demo/main.bal

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/websocket;
+import websocket_test.types;
+import ballerina/io;
+
+listener websocket:Listener websocketListener = check new(9092);
+map<types:User> users = {};
+
+@websocket:ServiceConfig{dispatcherKey: "event"}
+service / on websocketListener {
+    # An echo service that echoes the messages sent by the client.
+    # + return - User status
+    resource function get .() returns websocket:Service|websocket:UpgradeError {
+        return new WsService();
+    }
+
+}
+
+@websocket:ServiceConfig{dispatcherKey: "event"}
+service /user on websocketListener {
+    # Allows clients to get real-time data on users and chat with them.
+    # + return - websocket service
+    resource function get .() returns websocket:Service|websocket:UpgradeError {
+        return new WsServiceUser();
+    }
+
+}
+
+service class WsService {
+    *websocket:Service;
+
+    remote function onHello(types:Hello clientData) returns types:Response? {
+        return {message:"You sent: " + clientData.message};
+    }
+
+}
+
+service class WsServiceUser {
+    *websocket:Service;
+    
+    remote function onSubscribe(websocket:Caller caller, types:Subscribe sub) returns stream<types:Response>|error {
+        string callerId = caller.getConnectionId();
+        io:println(sub.name + " subscribed: " + callerId);
+        types:User user = {caller: caller, gender: sub.gender, name: sub.name, id: callerId};
+        users[callerId] = user;
+        broadcast("System: User " + user.name + " (" + callerId + ")" + " has joined the chat");
+        types:Response[] response = [{message: "System: Welcome to the chat!", event:"subscribe"}];
+        return response.toStream();
+    } 
+
+    remote function onUnsubscribe(websocket:Caller caller, types:Unsubscribe unsubscribe) returns error? {
+        string callerId = caller.getConnectionId();
+        broadcast("System: User " + users.get(callerId).name + " has left the chat");
+        _ = users.remove(callerId);
+    }
+
+    // remote function onClose(websocket:Caller caller) returns websocket:Error? {
```

Comment:
Any particular reason for commenting out these lines? I think it's better if we have this part as well. To remove the users from the map when the connection is closed by the client.

Reply:
The reason for this is that AsyncAPI tools currently doesn't support built-in functions in the `ballerina/websocket` library.

Reply:
Yes, but according to this function [1], we are explicitly disallowing built-in methods in the `websocket service`. This may be due to the difficulty of mapping the responses of these functions. 

[1] [https://github.com/hasathcharu/asyncapi-tools/blob/605bffe350b8f3d8c7f475184ffd201d33b4d4bf/asyncapi-cli/src/main/java/io/ballerina/asyncapi/websocketscore/generators/asyncspec/service/AsyncApiRemoteMapper.java#L249C1-L254C6](https://github.com/hasathcharu/asyncapi-tools/blob/605bffe350b8f3d8c7f475184ffd201d33b4d4bf/asyncapi-cli/src/main/java/io/ballerina/asyncapi/websocketscore/generators/asyncspec/service/AsyncApiRemoteMapper.java#L249C1-L254C6)

Reply:
Ohh, this is wrong. We should not throw any errors if those functions are there, we basically have to skip(ignore) them when generating the Async API. Will create an issue for this. Thanks for explaining.